src/modules/details/details.hook.ts

```diff
@@ -1,32 +1,72 @@
-import { useQuery } from 'react-query';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 import { toast } from 'react-toastify';
+import { AxiosError } from 'axios';
 
 import API_CONSTANTS from 'hooks/constants';
-import { getDepartmentList, getDetailsTicket } from './details.service';
+import {
+  getDetailsTicket,
+  putEditTicket,
+  putReopenTicket,
+} from './details.service';
+import { IEditTicketPayload, IReopenTicketPayload } from './type';
+import { ICreateTicketError } from 'modules/Ticket/type';
 
-export const useDetails = (id: number) => {
-  const { data, isLoading, refetch } = useQuery(
-    [API_CONSTANTS.DETAILS_SPECEFIC, id],
-    () => getDetailsTicket(id),
+export const useEditTicket = () => {
+  const queryClient = useQueryClient();
+  return useMutation(
+    ({ id, ticket }: { id: number; ticket: { ticket: IEditTicketPayload } }) =>
+      putEditTicket({ id, ticket }),
+    {
+      onSuccess: (res) => {
+        toast.success(res?.data?.message);
+        queryClient.invalidateQueries([API_CONSTANTS.DETAILS_SPECEFIC]);
+      },
+      onError: (err: AxiosError) => {
+        let error = err?.response?.data as ICreateTicketError;
+        toast.error(error?.errors || 'Failed to create ticket.');
+      },
+    }
+  );
+};
+
+export const useReopenTicket = () => {
+  const queryClient = useQueryClient();
+  return useMutation(
+    ({
+      id,
+      ticket_result,
+    }: {
+      id: number;
+      ticket_result: { ticket_result: IReopenTicketPayload };
+    }) => putReopenTicket({ id, ticket_result }),
     {
-      onError: (err) => {
+      onSuccess: (res) => {
+        toast.success(res?.data?.message);
+        queryClient.invalidateQueries([API_CONSTANTS.DETAILS_SPECEFIC]);
+      },
```

Comment:
toast me default value honi chaiye nahi to undefine ke time empty toast generate hoga